<html><head><meta http-equiv="X-UA-Compatible" content="IE=edge" /><link rel="shortcut icon" href="../icons/favicon.ico" /><link rel="stylesheet" type="text/css" href="../styles/branding.css" /><link rel="stylesheet" type="text/css" href="../styles/branding-en-US.css" /><script type="text/javascript" src="../scripts/branding.js"> </script><meta http-equiv="Content-Type" content="text/html; charset=UTF-8" /><title>Bio.Algorithms.Alignment Namespace</title><meta name="Language" content="en-us" /><meta name="System.Keywords" content="Bio.Algorithms.Alignment namespace" /><meta name="Microsoft.Help.F1" content="Bio.Algorithms.Alignment" /><meta name="Microsoft.Help.Id" content="N:Bio.Algorithms.Alignment" /><meta name="Description" content="summaryN:Bio.Algorithms.Alignment" /><meta name="Microsoft.Help.ContentType" content="Reference" /><meta name="BrandingAware" content="true" /><meta name="container" content="Bio.Algorithms.Alignment" /><meta name="file" content="15f500b8-4f6c-af6e-100f-4b9aa389ded8" /><meta name="guid" content="15f500b8-4f6c-af6e-100f-4b9aa389ded8" /><link rel="stylesheet" type="text/css" href="../styles/branding-Website.css" /><script type="text/javascript" src="../scripts/jquery-1.11.0.min.js"></script><script type="text/javascript" src="../scripts/branding-Website.js"></script></head><body onload="OnLoad('cs')"><input type="hidden" id="userDataCache" class="userDataStyle" /><div class="pageHeader" id="PageHeader">.NET Bio<form id="SearchForm" method="get" action="#" onsubmit="javascript:TransferToSearchPage(); return false;"><input id="SearchTextBox" type="text" maxlength="200" /><button id="SearchButton" type="submit"></button></form></div><div class="pageBody"><div class="leftNav" id="leftNav"><div id="tocNav"><div class="toclevel0" data-toclevel="0"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="9836064c-7064-aaa0-909a-aee4043a25aa.htm" title=".NET Bio" tocid="roottoc">.NET Bio</a></div><div class="toclevel1 current" data-toclevel="1" data-childrenloaded="true"><a class="tocExpanded" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="15f500b8-4f6c-af6e-100f-4b9aa389ded8.htm" title="Bio.Algorithms.Alignment" tocid="15f500b8-4f6c-af6e-100f-4b9aa389ded8">Bio.Algorithms.Alignment</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="03ae29ff-05fa-425c-8820-5c7a93be4918.htm" title="AlignedSequence Class" tocid="03ae29ff-05fa-425c-8820-5c7a93be4918">AlignedSequence Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="2baa377a-1277-455d-91a8-aeae647c0e44.htm" title="AlignmentInfo Class" tocid="2baa377a-1277-455d-91a8-aeae647c0e44">AlignmentInfo Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="2a3c9ae3-64d3-5419-629a-51424f204353.htm" title="Cluster Class" tocid="2a3c9ae3-64d3-5419-629a-51424f204353">Cluster Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="ad3b7a00-14e0-21e8-23cc-e45518e5f00c.htm" title="ClusterBuilder Class" tocid="ad3b7a00-14e0-21e8-23cc-e45518e5f00c">ClusterBuilder Class</a></div><div class="toclevel2" data-toclevel="2"><a data-tochassubtree="false" href="59168c86-ecc9-6b57-d039-870758bfb8d2.htm" title="ClusterScoreMethod Enumeration" tocid="59168c86-ecc9-6b57-d039-870758bfb8d2">ClusterScoreMethod Enumeration</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="78df2b94-f24c-8152-ed6c-c882bf375bee.htm" title="DeltaAlignment Class" tocid="78df2b94-f24c-8152-ed6c-c882bf375bee">DeltaAlignment Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="653c904d-8694-cdfe-0de9-1c284aae47d7.htm" title="DynamicProgrammingPairwiseAligner Class" tocid="653c904d-8694-cdfe-0de9-1c284aae47d7">DynamicProgrammingPairwiseAligner Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="f5ad2ed3-a5d1-8bf4-4aa9-552b6faf2b20.htm" title="DynamicProgrammingPairwiseAligner.OptScoreCell Structure" tocid="f5ad2ed3-a5d1-8bf4-4aa9-552b6faf2b20">DynamicProgrammingPairwiseAligner.OptScoreCell Structure</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="7822aa88-029d-f33e-30b8-e5b140e02d06.htm" title="DynamicProgrammingPairwiseAlignerJob Class" tocid="7822aa88-029d-f33e-30b8-e5b140e02d06">DynamicProgrammingPairwiseAlignerJob Class</a></div><div class="toclevel2" data-toclevel="2"><a data-tochassubtree="false" href="f119a46c-cb58-3132-2eef-9a19ca7b15bd.htm" title="DynamicProgrammingPairwiseAlignerJob.WeightFunction Delegate" tocid="f119a46c-cb58-3132-2eef-9a19ca7b15bd">DynamicProgrammingPairwiseAlignerJob.WeightFunction Delegate</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="11e2ee2f-fe61-a28d-5da0-b50ad3798fb4.htm" title="IAlignedSequence Interface" tocid="11e2ee2f-fe61-a28d-5da0-b50ad3798fb4">IAlignedSequence Interface</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="7506df87-a0b4-0086-cb34-51ebb0584344.htm" title="IAlignmentAttributes Interface" tocid="7506df87-a0b4-0086-cb34-51ebb0584344">IAlignmentAttributes Interface</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="5085e65e-c62e-9d2b-59b2-1178d2b9b126.htm" title="IClusterBuilder Interface" tocid="5085e65e-c62e-9d2b-59b2-1178d2b9b126">IClusterBuilder Interface</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="b61eb07a-7449-5ed9-f8e5-232fea6d303f.htm" title="IMultipleSequenceAligner Interface" tocid="b61eb07a-7449-5ed9-f8e5-232fea6d303f">IMultipleSequenceAligner Interface</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="9caac518-2a7f-ca55-4f2c-6ef19fe3b932.htm" title="IPairwiseSequenceAligner Interface" tocid="9caac518-2a7f-ca55-4f2c-6ef19fe3b932">IPairwiseSequenceAligner Interface</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="4292dd4f-e81b-1e84-d5d1-7d17ee34bf08.htm" title="IPairwiseSequenceAlignment Interface" tocid="4292dd4f-e81b-1e84-d5d1-7d17ee34bf08">IPairwiseSequenceAlignment Interface</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="1180f5f3-cb05-5477-6d97-1ad2a7c40a01.htm" title="ISequenceAligner Interface" tocid="1180f5f3-cb05-5477-6d97-1ad2a7c40a01">ISequenceAligner Interface</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="5441c0fc-fc30-43a4-8fa6-db27576a66f7.htm" title="ISequenceAlignment Interface" tocid="5441c0fc-fc30-43a4-8fa6-db27576a66f7">ISequenceAlignment Interface</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="741df0dd-345a-4ed1-9e1a-56d8c82c28ba.htm" title="ModifiedSmithWaterman Class" tocid="741df0dd-345a-4ed1-9e1a-56d8c82c28ba">ModifiedSmithWaterman Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="5620a6ed-7c1d-a8b0-caad-29d3de289510.htm" title="MultiSequenceAlignment Class" tocid="5620a6ed-7c1d-a8b0-caad-29d3de289510">MultiSequenceAlignment Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="a469e50a-7c49-3465-cc18-0bf9be3cec3b.htm" title="NeedlemanWunschAligner Class" tocid="a469e50a-7c49-3465-cc18-0bf9be3cec3b">NeedlemanWunschAligner Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="ce4f5799-63c9-4de5-a784-2f8883f30974.htm" title="NUCmer Class" tocid="ce4f5799-63c9-4de5-a784-2f8883f30974">NUCmer Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="34855c57-ba3c-9aab-d238-a401ec9952d7.htm" title="NUCmerAttributes Class" tocid="34855c57-ba3c-9aab-d238-a401ec9952d7">NUCmerAttributes Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="aa130d59-1faf-8445-cbc9-ad146c03afd0.htm" title="NucmerPairwiseAligner Class" tocid="aa130d59-1faf-8445-cbc9-ad146c03afd0">NucmerPairwiseAligner Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="a6e3f67f-8ec0-e737-f03a-ddbe6e3875bf.htm" title="PairwiseAlignedSequence Class" tocid="a6e3f67f-8ec0-e737-f03a-ddbe6e3875bf">PairwiseAlignedSequence Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="ad670215-1f99-94e6-3405-e798e030948f.htm" title="PairwiseAlignmentAttributes Class" tocid="ad670215-1f99-94e6-3405-e798e030948f">PairwiseAlignmentAttributes Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="e1b641cd-d66f-d18f-94de-489d2e50673c.htm" title="PairwiseOverlapAffineAlignmentJob Class" tocid="e1b641cd-d66f-d18f-94de-489d2e50673c">PairwiseOverlapAffineAlignmentJob Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="dabd101a-0069-6908-db19-1cf875db1b61.htm" title="PairwiseOverlapAligner Class" tocid="dabd101a-0069-6908-db19-1cf875db1b61">PairwiseOverlapAligner Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="411fcb76-16e0-8dbc-bddb-885d303dd16e.htm" title="PairwiseOverlapSimpleAlignmentJob Class" tocid="411fcb76-16e0-8dbc-bddb-885d303dd16e">PairwiseOverlapSimpleAlignmentJob Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="7ebb8308-386a-01c8-9fbf-44bfa6e51384.htm" title="PairwiseSequenceAligner Class" tocid="7ebb8308-386a-01c8-9fbf-44bfa6e51384">PairwiseSequenceAligner Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="21d374a0-efba-df50-ab70-f7fd26b06262.htm" title="PairwiseSequenceAligner.OptScoreMatrixCell Class" tocid="21d374a0-efba-df50-ab70-f7fd26b06262">PairwiseSequenceAligner.OptScoreMatrixCell Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="90acb30e-68f6-9f74-b0f7-fb29f43dfe38.htm" title="PairwiseSequenceAlignment Class" tocid="90acb30e-68f6-9f74-b0f7-fb29f43dfe38">PairwiseSequenceAlignment Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="2e66d201-589a-8719-ca55-0d43c1aa6d10.htm" title="SequenceAligners Class" tocid="2e66d201-589a-8719-ca55-0d43c1aa6d10">SequenceAligners Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="323429e4-809e-c1c9-89f7-c42d49b784a7.htm" title="SequenceAlignment Class" tocid="323429e4-809e-c1c9-89f7-c42d49b784a7">SequenceAlignment Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="7da9dc8a-dec6-a52d-195a-62b6d4dff40c.htm" title="SmithWatermanAligner Class" tocid="7da9dc8a-dec6-a52d-195a-62b6d4dff40c">SmithWatermanAligner Class</a></div><div class="toclevel2" data-toclevel="2"><a class="tocCollapsed" onclick="javascript: Toggle(this);" href="#!" /><a data-tochassubtree="true" href="46585a1b-ce46-2b0d-2f0a-62f41199e38f.htm" title="Synteny Class" tocid="46585a1b-ce46-2b0d-2f0a-62f41199e38f">Synteny Class</a></div></div><div id="tocResizableEW" onmousedown="OnMouseDown(event);"></div><div id="TocResize" class="tocResize"><img id="ResizeImageIncrease" src="../icons/TocOpen.gif" onclick="OnIncreaseToc()" alt="Click or drag to resize" title="Click or drag to resize" /><img id="ResizeImageReset" src="../icons/TocClose.gif" style="display:none" onclick="OnResetToc()" alt="Click or drag to resize" title="Click or drag to resize" /></div></div><div class="topicContent" id="TopicContent"><table class="titleTable"><tr><td class="titleColumn">Bio.Algorithms.Alignment Namespace</td></tr></table><span class="introStyle"></span><div class="summary"><p style="color: #dc143c; font-size: 8.5pt; font-weight: bold;">[Missing &lt;summary&gt; documentation for "N:Bio.Algorithms.Alignment"]</p></div><div class="collapsibleAreaRegion"><span class="collapsibleRegionTitle" onclick="SectionExpandCollapse('ID0RB')" onkeypress="SectionExpandCollapse_CheckKey('ID0RB', event)" tabindex="0"><img id="ID0RBToggle" class="collapseToggle" src="../icons/SectionExpanded.png" />Classes</span></div><div id="ID0RBSection" class="collapsibleSection"><table id="typeList" class="members"><tr><th class="iconColumn">
					 
				</th><th>Class</th><th>Description</th></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="03ae29ff-05fa-425c-8820-5c7a93be4918.htm">AlignedSequence</a></td><td><div class="summary">
            AlignedSequence is a class containing the single aligned unit of alignment.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="2baa377a-1277-455d-91a8-aeae647c0e44.htm">AlignmentInfo</a></td><td><div class="summary">
            This class defines generic properties of any alignment algorithm.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="2a3c9ae3-64d3-5419-629a-51424f204353.htm">Cluster</a></td><td><div class="summary">
            An ordered list of matches between two a pair of sequences
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="ad3b7a00-14e0-21e8-23cc-e45518e5f00c.htm">ClusterBuilder</a></td><td><div class="summary">
            Clustering is a process in which individual matches are grouped in larger
            set called Cluster. The matches in cluster are decided based on paramters 
            like fixed difference allowed, maximum difference allowed, minimum score
            and separation factor that should be satisfied.
            This class implements IClusterBuilder interface.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="78df2b94-f24c-8152-ed6c-c882bf375bee.htm">DeltaAlignment</a></td><td><div class="summary">
            Represents a alignment object in terms of delta.
            Delta is an encoded representation of alignments between input sequences.
            It contains the start and end indices of alignment in reference and
            query sequence followed by error values and list of integer in 
            following lines. Each integer represent an insertion (+ve) in reference
            sequence and deletion (-ve) in reference sequence.
            This class represents such alignment with required properties and
            utility methods.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="653c904d-8694-cdfe-0de9-1c284aae47d7.htm">DynamicProgrammingPairwiseAligner</a></td><td><div class="summary">
            Base class for dynamic programming alignment algorithms, including NeedlemanWunsch, 
            SmithWaterman and PairwiseOverlap.
            The basic reference for this code (including NW, SW and Overlap) is Chapter 2 in 
            Biological Sequence Analysis; Durbin, Eddy, Krogh and Mitchison; Cambridge Press; 1998
            The variable names in these classes follow the notation in Durbin et al.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="7822aa88-029d-f33e-30b8-e5b140e02d06.htm">DynamicProgrammingPairwiseAlignerJob</a></td><td><div class="summary">
            Base class for all aligner algorithms.
            Provides storage and read/write operations for the grid.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="741df0dd-345a-4ed1-9e1a-56d8c82c28ba.htm">ModifiedSmithWaterman</a></td><td><div class="summary">
            Implements algorithm to extend given pair of sequences in
            specified direction. This implementation is specific to NUCmer.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="5620a6ed-7c1d-a8b0-caad-29d3de289510.htm">MultiSequenceAlignment</a></td><td><div class="summary" /></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="a469e50a-7c49-3465-cc18-0bf9be3cec3b.htm">NeedlemanWunschAligner</a></td><td><div class="summary">
            Implements the NeedlemanWunsch algorithm for global alignment.
            See Chapter 2 in Biological Sequence Analysis; Durbin, Eddy, Krogh and Mitchison; 
            Cambridge Press; 1998.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="ce4f5799-63c9-4de5-a784-2f8883f30974.htm">NUCmer</a></td><td><div class="summary">
            NUCmer is a system for rapidly aligning entire genomes or very large DNA
            sequences. It allows alignment of multiple reference sequences to multiple query sequences.
            This is commonly used to identify the position and orientation of set of sequence
            contigs in relation to finished sequence.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="34855c57-ba3c-9aab-d238-a401ec9952d7.htm">NUCmerAttributes</a></td><td><div class="summary">
            This class extends MUMmerAttributes and adds NUCmer specific attributes
            required to run the NUCmer algorithm.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="aa130d59-1faf-8445-cbc9-ad146c03afd0.htm">NucmerPairwiseAligner</a></td><td><div class="summary">
            Class which uses nucmer to perform an alignment.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="a6e3f67f-8ec0-e737-f03a-ddbe6e3875bf.htm">PairwiseAlignedSequence</a></td><td><div class="summary">
            PairwiseAlignedSequence is a class containing the single aligned unit of pairwise alignment.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="ad670215-1f99-94e6-3405-e798e030948f.htm">PairwiseAlignmentAttributes</a></td><td><div class="summary">
            This class implements IAlignmentAttributes interface and defines all the 
            parameters required to run any pairwise algorithm.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="e1b641cd-d66f-d18f-94de-489d2e50673c.htm">PairwiseOverlapAffineAlignmentJob</a></td><td><div class="summary">
            Pairwise Overlap alignment implementation using affine gap model.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="dabd101a-0069-6908-db19-1cf875db1b61.htm">PairwiseOverlapAligner</a></td><td><div class="summary">
            Implements the pair-wise overlap alignment algorithm described in Chapter 2 of
            Biological Sequence Analysis; Durbin, Eddy, Krogh and Mitchison; Cambridge Press; 1998.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="411fcb76-16e0-8dbc-bddb-885d303dd16e.htm">PairwiseOverlapSimpleAlignmentJob</a></td><td><div class="summary">
            Pairwise Overlap alignment implementation using simple gap model.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="7ebb8308-386a-01c8-9fbf-44bfa6e51384.htm">PairwiseSequenceAligner</a></td><td><div class="summary">
            Base class for our pair-wise sequence aligners. This implements the core shared 
            portions of the Smith-Waterman and Needleman-Wunsch aligners.
            </div></td></tr><tr data="class; public"><td><img src="../icons/protclass.gif" alt="Protected class" title="Protected class" /></td><td><a href="21d374a0-efba-df50-ab70-f7fd26b06262.htm">PairwiseSequenceAligner<span id="LSTD84D45E3_0"></span><script type="text/javascript">AddLanguageSpecificTextSet("LSTD84D45E3_0?cs=.|vb=.|cpp=::|nu=.|fs=.");</script>OptScoreMatrixCell</a></td><td><div class="summary">
            Optimum score maxtrix cell
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="90acb30e-68f6-9f74-b0f7-fb29f43dfe38.htm">PairwiseSequenceAlignment</a></td><td><div class="summary">
            A simple implementation of IPairwiseSequenceAlignment that stores the 
            results as list of Aligned Sequences.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="2e66d201-589a-8719-ca55-0d43c1aa6d10.htm">SequenceAligners</a></td><td><div class="summary">
            SequenceAligners class is an abstraction class which provides instances
            and lists of all Aligners currently supported by Bio. 
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="323429e4-809e-c1c9-89f7-c42d49b784a7.htm">SequenceAlignment</a></td><td><div class="summary">
            A simple implementation of ISequenceAlignment that stores the 
            result of an alignment. 
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="7da9dc8a-dec6-a52d-195a-62b6d4dff40c.htm">SmithWatermanAligner</a></td><td><div class="summary">
            Implements the SmithWaterman algorithm for partial alignment.
            See Chapter 2 in Biological Sequence Analysis; Durbin, Eddy, Krogh and Mitchison; 
            Cambridge Press; 1998.
            </div></td></tr><tr data="class; public"><td><img src="../icons/pubclass.gif" alt="Public class" title="Public class" /></td><td><a href="46585a1b-ce46-2b0d-2f0a-62f41199e38f.htm">Synteny</a></td><td><div class="summary">
            An ordered list of clusters between two sequences A and B
            </div></td></tr></table></div><div class="collapsibleAreaRegion"><span class="collapsibleRegionTitle" onclick="SectionExpandCollapse('ID1RB')" onkeypress="SectionExpandCollapse_CheckKey('ID1RB', event)" tabindex="0"><img id="ID1RBToggle" class="collapseToggle" src="../icons/SectionExpanded.png" />Structures</span></div><div id="ID1RBSection" class="collapsibleSection"><table id="typeList" class="members"><tr><th class="iconColumn">
					 
				</th><th>Structure</th><th>Description</th></tr><tr data="structure; public"><td><img src="../icons/protstructure.gif" alt="Protected structure" title="Protected structure" /></td><td><a href="f5ad2ed3-a5d1-8bf4-4aa9-552b6faf2b20.htm">DynamicProgrammingPairwiseAligner<span id="LSTD84D45E3_1"></span><script type="text/javascript">AddLanguageSpecificTextSet("LSTD84D45E3_1?cs=.|vb=.|cpp=::|nu=.|fs=.");</script>OptScoreCell</a></td><td><div class="summary">
            Position details of cell with best score.
            </div></td></tr></table></div><div class="collapsibleAreaRegion"><span class="collapsibleRegionTitle" onclick="SectionExpandCollapse('ID2RB')" onkeypress="SectionExpandCollapse_CheckKey('ID2RB', event)" tabindex="0"><img id="ID2RBToggle" class="collapseToggle" src="../icons/SectionExpanded.png" />Interfaces</span></div><div id="ID2RBSection" class="collapsibleSection"><table id="typeList" class="members"><tr><th class="iconColumn">
					 
				</th><th>Interface</th><th>Description</th></tr><tr data="interface; public"><td><img src="../icons/pubinterface.gif" alt="Public interface" title="Public interface" /></td><td><a href="11e2ee2f-fe61-a28d-5da0-b50ad3798fb4.htm">IAlignedSequence</a></td><td><div class="summary">
            Interface to hold single aligned unit of alignment.
            </div></td></tr><tr data="interface; public"><td><img src="../icons/pubinterface.gif" alt="Public interface" title="Public interface" /></td><td><a href="7506df87-a0b4-0086-cb34-51ebb0584344.htm">IAlignmentAttributes</a></td><td><div class="summary">
            Aligners will implements this interface to list the attributes supported
            or required.
            </div></td></tr><tr data="interface; public"><td><img src="../icons/pubinterface.gif" alt="Public interface" title="Public interface" /></td><td><a href="5085e65e-c62e-9d2b-59b2-1178d2b9b126.htm">IClusterBuilder</a></td><td><div class="summary">
            Contract defined to implement class that creates clusters.
            Takes list of maximum unique matches as input and creates clusters
            </div></td></tr><tr data="interface; public"><td><img src="../icons/pubinterface.gif" alt="Public interface" title="Public interface" /></td><td><a href="b61eb07a-7449-5ed9-f8e5-232fea6d303f.htm">IMultipleSequenceAligner</a></td><td><div class="summary">
            Multiple sequences alignment interface.
            Multiple sequence alignment (MSA) is used to align three or  
            more sequences in preparation for further analysis.  
            More info on MSA can be found at 
            http://en.wikipedia.org/wiki/Multiple_sequence_alignment)
            </div></td></tr><tr data="interface; public"><td><img src="../icons/pubinterface.gif" alt="Public interface" title="Public interface" /></td><td><a href="9caac518-2a7f-ca55-4f2c-6ef19fe3b932.htm">IPairwiseSequenceAligner</a></td><td><div class="summary">
            A sequence alignment algorithm that aligns exactly two 
            sequences. This may diverge from ISequenceAligner at some 
            point; meanwhile, it's important to maintain the distinction
            (e.g., assembly requires a pairwise algorithm).
            </div></td></tr><tr data="interface; public"><td><img src="../icons/pubinterface.gif" alt="Public interface" title="Public interface" /></td><td><a href="4292dd4f-e81b-1e84-d5d1-7d17ee34bf08.htm">IPairwiseSequenceAlignment</a></td><td><div class="summary">
            An IPairwiseSequenceAlignment is the result of running a Pairwise alignment algorithm on a set 
            of two sequences.
            </div></td></tr><tr data="interface; public"><td><img src="../icons/pubinterface.gif" alt="Public interface" title="Public interface" /></td><td><a href="1180f5f3-cb05-5477-6d97-1ad2a7c40a01.htm">ISequenceAligner</a></td><td><div class="summary">
            A sequence aligner is an algorithm which takes N sequences as input and produces an 
            alignment of the sequences as output.
            </div></td></tr><tr data="interface; public"><td><img src="../icons/pubinterface.gif" alt="Public interface" title="Public interface" /></td><td><a href="5441c0fc-fc30-43a4-8fa6-db27576a66f7.htm">ISequenceAlignment</a></td><td><div class="summary">
            An ISequenceAlignment is the result of running an alignment algorithm on a set 
            of two or more sequences. This could be a pairwise alignment, an MSA (multiple 
            sequence alignment), or an overlap alignment of the sort needed for sequence
            assembly.
            </div></td></tr></table></div><div class="collapsibleAreaRegion"><span class="collapsibleRegionTitle" onclick="SectionExpandCollapse('ID3RB')" onkeypress="SectionExpandCollapse_CheckKey('ID3RB', event)" tabindex="0"><img id="ID3RBToggle" class="collapseToggle" src="../icons/SectionExpanded.png" />Delegates</span></div><div id="ID3RBSection" class="collapsibleSection"><table id="typeList" class="members"><tr><th class="iconColumn">
					 
				</th><th>Delegate</th><th>Description</th></tr><tr data="delegate; public"><td><img src="../icons/protdelegate.gif" alt="Protected delegate" title="Protected delegate" /></td><td><a href="f119a46c-cb58-3132-2eef-9a19ca7b15bd.htm">DynamicProgrammingPairwiseAlignerJob<span id="LSTD84D45E3_2"></span><script type="text/javascript">AddLanguageSpecificTextSet("LSTD84D45E3_2?cs=.|vb=.|cpp=::|nu=.|fs=.");</script>WeightFunction</a></td><td><div class="summary">
            Algorith-specific weight function
            </div></td></tr></table></div><div class="collapsibleAreaRegion"><span class="collapsibleRegionTitle" onclick="SectionExpandCollapse('ID4RB')" onkeypress="SectionExpandCollapse_CheckKey('ID4RB', event)" tabindex="0"><img id="ID4RBToggle" class="collapseToggle" src="../icons/SectionExpanded.png" />Enumerations</span></div><div id="ID4RBSection" class="collapsibleSection"><table id="typeList" class="members"><tr><th class="iconColumn">
					 
				</th><th>Enumeration</th><th>Description</th></tr><tr data="enumeration; public"><td><img src="../icons/pubenumeration.gif" alt="Public enumeration" title="Public enumeration" /></td><td><a href="59168c86-ecc9-6b57-d039-870758bfb8d2.htm">ClusterScoreMethod</a></td><td><div class="summary">
            Enum to indicate method to use while calculating score of a cluster.
            </div></td></tr></table></div></div></div><div id="pageFooter" class="pageFooter"> </div></body></html>